pkg/analyzer/analyzer.go

```diff
@@ -41,46 +81,116 @@ func newVisitor(pass *analysis.Pass) func(node ast.Node) {
 			return
 		}
 
-		var outgoingFieldList *ast.FieldList
+		funcName := a.funcName(deferStmt.Call)
+		if funcName != "" && a.exclude.MatchesAny(funcName) {
+			return
+		}
 
+		var isFuncReturnType bool
```

Comment:
Shoulnd it be called `hasReturn`? We're checking here if statement returns anything, not the type.